File: backend/app/schemas/watchlist.py
from pydantic import BaseModel


class WatchlistItem(BaseModel):
    symbol: str
    company_name: str
    company_name_ko: str | None
    market_type: str
    sector: str | None
    price_local: float | None
    currency: str
    price_krw: float | None
    daily_change_percent: float | None
    daily_change_amount_local: float | None
    daily_change_amount_krw: float | None
    session_status: str
    market_cap: float | None
    volume: float | None
    fx_impact_percent: float | None


class WatchlistResponse(BaseModel):
    items: list[WatchlistItem]
    usdkrw: float | None


class InstrumentSearchItem(BaseModel):
    symbol: str
    company_name: str
    company_name_ko: str | None
    market_type: str
    sector: str | None
    in_watchlist: bool

File: frontend/src/types/watchlist.ts
export interface WatchlistItem {
  symbol: string;
  company_name: string;
  company_name_ko: string | null;
  market_type: "KOR" | "US";
  sector: string | null;
  price_local: number | null;
  currency: string;
  price_krw: number | null;
  daily_change_percent: number | null;
  daily_change_amount_local: number | null;
  daily_change_amount_krw: number | null;
  session_status: "open" | "closed" | "pre_market" | "after_market";
  market_cap: number | null;
  volume: number | null;
  fx_impact_percent: number | null;
}

export interface WatchlistResponse {
  items: WatchlistItem[];
  usdkrw: number | null;
}

export interface InstrumentSearchItem {
  symbol: string;
  company_name: string;
  company_name_ko: string | null;
  market_type: "KOR" | "US";
  sector: string | null;
  in_watchlist: boolean;
}

export type WatchlistSortKey = "display_order" | "change" | "krw_delta" | "market_cap";
export type MarketFilter = "all" | "kor" | "us";
